core-httpclient-impl/src/main/java/com/optimizely/ab/OptimizelyFactory.java

```diff
@@ -221,7 +230,16 @@ public static Optimizely newDefaultInstance(String sdkKey) {
             return newDefaultInstance(() -> null);
         }
 
-        return newDefaultInstance(sdkKey, null);
+        return newDefaultInstance(sdkKey, "");
+    }
+
+    public static Optimizely newDefaultInstance(String sdkKey, CloseableHttpClient httpClient) {
```

Comment:
I don't think so we need this method.

Reply:
What about removing this and add httpClient option to "newDefaultInstance(String sdkKey, String fallback, String datafileAccessToken, ClosableHttpClient httpClient)? Then no need for extra set method. Considering demands for customized httpclient, it may be reasonable.

Reply:
we can add default null in newDefaultInstance but that wouldn't be easy for the customer to initialize using your suggested method.